server.js

```diff
@@ -1,38 +1,258 @@
-import express from 'express'
-import bodyParser from 'body-parser'
-import cors from 'cors'
-import mongoose from 'mongoose'
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
-mongoose.Promise = Promise
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
+import express from 'express';
+import bodyParser from 'body-parser';
+import cors from 'cors';
+import mongoose from 'mongoose';
+import booksData from './data/books.json';
+import authorsData from './data/authors.json';
+
+const mongoUrl = process.env.MONGO_URL || 'mongodb://localhost/project-mongo';
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.Promise = Promise;
+
+// Create my database model's
+const Book = mongoose.model('Book', {
+  bookID: Number,
+  title: String,
+  authors: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'Author'
+  },
+  average_rating: Number,
+  isbn: String,
+  isbn13: String,
+  language_code: String,
+  num_pages: Number,
+  ratings_count: Number,
+  text_reviews_count: Number
+});
+
+const Author = mongoose.model('Author', {
+  authors: String
+});
+
+if (process.env.RESET_DATABASE) {
+  // POPULATING DATABASE WITH TWO COLLECTIONS (WITH RELATIONS)
+  const populateDatabase = async () => {
+    // First of all, clear current content of two collections
+    await Book.deleteMany();
+    await Author.deleteMany();
+
+    // Next, declare empty array in which later on will
+    // store all authors from authors.json (of Author models)
+    let authorsArray = [];
+
+    authorsData.forEach(async (item) => {
+      const newAuthor = new Author(item);
+
+      // Push each newAuthor to array authorsArray
+      authorsArray.push(newAuthor);
+      await newAuthor.save();
+    });
```

Comment:
This is very clear and I like clarity. Thank you for writing these comments 👍 
It's interesting that we did slightly different in our code here. You've managed to achieve the same population-purpose as I did, but in less code than I used. I will have to take a second look at my code to see if I maybe should update it a bit.. (I'm using both map and forEach) 🤔